### DISCUSSION TITLE
Issue using this method when launching from Shell

### DISCUSSION THREAD

**Discussion by devonuto** [brminnick/HackerNews#33] [General]
I took this example and implemented an MVVM model, but I launch the page from Shell under a flyout item as shellcontent like:

new ShellContent
{
    Title = "SUMMARY",
    Route = nameof(DashboardPage),
    ContentTemplate = new DataTemplate(typeof(DashboardPage)),
},

And when launching, I now get a "System.MissingMethodException: 'No parameterless constructor defined for type DashboardPage" error, as it appears to be expecting the viewmodel as a parameter. Not sure how that can be done?

**Comment by TheCodeTraveler** [comment]
Hi @devontu,
It sounds like you've found a bug in .NET MAUI. I recommend opening an Issue on the .NET MAUI repo: https://github.com/dotnet/maui
